presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixPlugin.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.phoenix;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.spi.Plugin;
+import io.prestosql.spi.connector.ConnectorFactory;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+
+public class PhoenixPlugin
+        implements Plugin
+{
+    @Override
+    public Iterable<ConnectorFactory> getConnectorFactories()
+    {
+        return ImmutableList.of(new PhoenixConnectorFactory(getClassLoader()));
+    }
+
+    private static ClassLoader getClassLoader()
+    {
+        return firstNonNull(Thread.currentThread().getContextClassLoader(), PhoenixPlugin.class.getClassLoader());
```

Comment:
Remove this method and use `PhoenixClient.class.getClassLoader()` above. That one is always correct -- we shouldn't rely on the thread context class loader, which is intended for other purposes. (we only set it for interoperability with Hadoop code)